Agenda
------

These are the lectures I'll give:

* Brief review of some topics covered in EN 600.439/639.
* Introduction to RNA sequencing data analysis
* Introduction to big-data methods for sequencing data analysis
* Project ideas

That's it!  The rest of the discussion is guided, fueled, and led by you.

Readings
--------

If you are taking my class and you have any trouble accessing these resources, please contact me. All of these articles should be easily accessible from the JHU campus or via VPN / library proxy.

### RNA sequencing data analysis

* Surveys (everyone should read these -- don't pick one of these to present in class):
    * [RNA-Seq: a revolutionary tool for transcriptomics](http://www.nature.com/nrg/journal/v10/n1/full/nrg2484.html) by Wang, Gerstein, Snyder
    * [Computational methods for transcriptome annotation and quantification using RNA-seq](http://www.nature.com/nmeth/journal/v8/n6/pdf/nmeth.1613.pdf) by Garber, Grabherr, Guttman and Trapnell
    * [From RNA-seq reads to differential expression results](http://www.biomedcentral.com/content/pdf/gb-2010-11-12-220.pdf) by Oshlack, Robinson, Young

* Spliced alignment
    * [MapSplice: Accurate mapping of RNA-seq reads for splice junction discovery](http://nar.oxfordjournals.org/content/38/18/e178.full.pdf) by Wang et al
    * [TopHat: discovering splice junctions with RNA-seq](http://bioinformatics.oxfordjournals.org/content/25/9/1105.full.pdf) by Trapnell, Pachter, Salzberg

* Assembly
    * Cufflinks
    * CLASS
    * Dewey

* Scalable methods
    * Myrna

### Big data methods for sequencing data analysis

* Indexing
    * NAR review
    * Navarro review
    * Bowtie
    * BWA
* Compression
    * CRAM
    * Quip
* Sketching and streaming
    * eXpress
* Scalable computing
    * Scalable alignment
        * CloudBurst
    * Scalable indexing
        * Menon et al
    * Scalable pipelines
